File: app/src/main/java/ai/elimu/content_provider/room/entity/Video.java
package ai.elimu.content_provider.room.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;

import ai.elimu.model.v2.enums.content.VideoFormat;

/**
 * For documentation, see https://github.com/elimu-ai/webapp/tree/main/src/main/java/ai/elimu/model
 */
@Entity
public class Video extends Content {

    @NonNull
    private String title;

    @NonNull
    private VideoFormat videoFormat;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public VideoFormat getVideoFormat() {
        return videoFormat;
    }

    public void setVideoFormat(VideoFormat videoFormat) {
        this.videoFormat = videoFormat;
    }
}

File: app/src/main/java/ai/elimu/content_provider/room/entity/Video.kt
package ai.elimu.content_provider.room.entity

import ai.elimu.model.v2.enums.content.VideoFormat
import androidx.room.Entity

/**
 * For documentation, see [model](https://github.com/elimu-ai/webapp/tree/main/src/main/java/ai/elimu/model)
 */
@Entity
class Video : Content() {
    @JvmField
    var title: String = ""

    lateinit var videoFormat: VideoFormat
}
